### ISSUE TITLE
Deprecation warnings on PHP 8.4

### ISSUE THREAD

**Issue by ADmad** [firebase/php-jwt#570]
```
src/JWT.php:199
Firebase\JWT\JWT::encode(): Implicitly marking parameter $keyId as nullable is deprecated, the explicit nullable type must be used instead

src/JWK.php:55
Firebase\JWT\JWK::parseKeySet(): Implicitly marking parameter $defaultAlg as nullable is deprecated, the explicit nullable type must be used instead

src/JWK.php:96
Firebase\JWT\JWK::parseKey(): Implicitly marking parameter $defaultAlg as nullable is deprecated, the explicit nullable type must be used instead
```

**Comment by lolli42** [comment]
Confirmed. Would be great to get this merged and released - PHP 8.4 is alpha2 now and frameworks start implementing compatibility, libraries should have compatible releases.

**Comment by lolli42** [comment]
Ping.

Note current CI is red, which is fixed by PR #572 as well as drive-by.

Note there is PR #573 as dupe already.
